sympy/functions/special/beta_functions.py

```diff
@@ -120,3 +121,81 @@ def _eval_conjugate(self):
 
     def _eval_rewrite_as_gamma(self, x, y, **kwargs):
         return self._eval_expand_func(**kwargs)
+
+
+###############################################################################
+########################### INCOMPLETE BETA  FUNCTION #########################
+###############################################################################
+
+
+
+class betainc(Function):
+    r"""
+    The incomplete beta function is a generalization of the beta function,
+
+    .. math::
+        \mathrm{B}_{z}(a,b) := \int^{z}_{0} t^{a-1} (1-t)^{b-1} \mathrm{d}t.
+
+    Incomplete beta function satisfies properties like:
+
+    .. math::
+        \mathrm{B}_{1}(a,b) = \mathrm{B}(a,b) \\
+
+    See Also
+    ========
+    Gamma function, Upper incomplete gamma function, Lower incomplete gamma function,
+    Polygamma function, Log Gamma function, Digamma function, Trigamma function.
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Beta_function#Incomplete_beta_function
+    .. [2] http://mathworld.wolfram.com/IncompleteBetaFunction.html
+    """
+    nargs = 3
+
+    def fdiff(self, argindex):
+        z, a, b = self.args
+        if argindex not in (1,2,3):
+            raise ArgumentIndexError(self,argindex)
+        elif argindex == 1:
+            # Diff wrt z
+            return (1 - z)**(b - 1)*z**(a - 1)
+        elif argindex == 2:
+            # Diff wrt a
+            term1 = betainc(z, a, b)*log(z)
+            term2 = z**a*gamma(a)**2*hyper([a, a, 1-b], [a+1, a+1], z)
+            return  term1 - term2
+        elif argindex == 3:
+            term1 = gamma(b)**2*(1 - z)**b*hyper([b, b, 1-a], [b+1, b+1], 1-z)
+            term2 = log(1 - z)*self.func(1 - z, a, b)
+            term3 = (digamma(b) - digamma(a + b))*beta(a, b)
+            return term1 - term2 + term3
+
+    @classmethod
+    def eval(cls, z, a, b):
+        #Case z = 0 , incomplete beta is zero
+        if z == 0:
+            return S.Zero
+        #Beta function is special case of Incomplete beta function for z=1
+        if z == 1:
+            return beta(a,b)
+
+    def _eval_rewrite_as_Integral(self,*args, **kwargs):
+        z, a, b = self.args
+        from sympy import Integral
+        t = Symbol('t')
+        integrand = t**(a - 1)*(1 - t)**(b - 1)
+        return Integral(integrand,(t, 0, z))
+
+    def _eval_rewrite_as_hyper(self,*args, **kwargs):
+        z, a, b = self.args
+        return gamma(a)*z**a*hyper([a, 1-b], [a+1], z)
+
+    def _eval_is_real(self):
+        z, a, b = self.args
+        return a.is_real and b.is_real and z.is_real
```

Comment:
The integrand has branch points at 0 and 1 (unless a and b are integers) so it is real only in the interval [0, 1], in general. Hence the test should be `z.is_nonnegative and (1 - z).is_nonnegative`. Strictly speaking, `fuzzy_and` should be used instead of `and` but that may not be important in practice.

Reply:
Yes. I missed that out.

Reply:
Please correct me if i am wrong:
1. If a and b are integers(+ve) then it depends on z being real or not.
2. if a and b are fractions(+ve) then it depends on z being in interval (0,1)
I am not sure about extension of inc beta fnc for -ve values of a and b.

Reply:
I think that it would suffice to return True for a, b > 0 and 0 <= z <= 1, and None otherwise. I would not try to find more extensive conditions as they may be complicated in general. The inequalities should be replaced by the corresponding properties `a.is_positive`, etc..

Reply:
Resolved and updated.